### PR TITLE
 Changed DNS operator status to use newer "config.openshift.io/v1" API version

### PR DESCRIPTION
**Pull request by pravisankar**
Depends on https://github.com/openshift/cluster-dns-operator/pull/41

Updated openshift-cluster-dns dependencies
- To utilize new cluster operator status API,
added dependency on 'github.com/openshift/api' and removed
dependency on 'github.com/openshift/cluster-version-operator'.

- Pinned operator-sdk to 'v0.0.6' as the existing code is not
compatible with new sdk versions.

- Now the ClusterOperator is cluster scoped and the operator status name is changed
from openshift-dns to openshift-cluster-dns-operator to comply with the new changes.

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge PTAL

**Comment by pravisankar** [comment]
/hold
Depends on https://github.com/openshift/origin/pull/21623

File: test/e2e/operator_test.go [hunk 1]
```diff
@@ -16,14 +16,13 @@ import (
...
+	co := &configv1.ClusterOperator{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "ClusterOperator",
-			APIVersion: "operatorstatus.openshift.io/v1",
+			APIVersion: "config.openshift.io/v1",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "openshift-dns",
-			Namespace: "openshift-cluster-dns-operator",
+			Name: "openshift-cluster-dns-operator",
```
[the remark this answers is not in the document]
Comment (ironcladlou) [reply test/e2e/operator_test.go]: Would a constant have caught this?

**Comment by pravisankar** [comment]
@openshift/sig-network-edge Updated, PTAL

**Comment by ironcladlou** [comment]
https://github.com/openshift/cluster-dns-operator/pull/41 merged, time to rebase this one

**Pull request merged**